### PR TITLE
Feat(vue): Adds SearchBox component to reactivesearch-vue lib

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** For ReactiveSearch, we will create a Searchbox component in ReactiveSearch instead of modifying the existing DataSearch component.

- [x]  Searchbox support in Vue

[🎥  Loom Vue](https://www.loom.com/share/4109b0f4e5534f3ba9e2b100bf18385f)

[Notion Link](https://www.notion.so/appbase/8648760a66da4dcfa5ebcefd2e439ddc)

**Related PR(s)**
- https://github.com/appbaseio/vue-playground/pull/37
- https://github.com/appbaseio/reactivecore/pull/115
- https://github.com/appbaseio/Docs/pull/265

**Snapshot Local Testing Status** ✅ 
image

### CODE REVIEW

**Comment by mohdashraf010897** [comment]
https://www.loom.com/share/b0d77f046648473795b17f2d93effe3a

File: packages/vue/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@appbaseio/reactivesearch-vue",
-	"version": "1.27.0",
+	"version": "1.27.0-gamma.2",
```
Comment (bietkul) [inline packages/vue/package.json]:
```suggestion
	"version": "1.27.0",
```

**Review by bietkul — APPROVED** [review]

**Review by bietkul — CHANGES_REQUESTED** [review]
@mohdashraf010897 There are following issues:
- Two API calls on each keystroke
- Suggestions dropdown is closed when user types in Input box, as a result it doesn't show suggestions
https://www.loom.com/share/faf46bf3f0db4914841cb2243bfc8654

**Comment by mohdashraf010897** [comment]
> @mohdashraf010897 There are following issues:
> 
> * Two API calls on each keystroke
> * Suggestions dropdown is closed when user types in Input box, as a result it doesn't show suggestions
>   [https://www.loom.com/share/faf46bf3f0db4914841cb2243bfc8654](https://www.loom.com/share/faf46bf3f0db4914841cb2243bfc8654)

@bietkul https://www.loom.com/share/ba527aa7fb2447938c75166cd6725110

**Comment by bietkul** [comment]
@mohdashraf010897 Update the example. Remove unnecessary props and keep the example simple.

**Review by bietkul — APPROVED** [review]

**Pull request merged**